sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/_connection_string_parser.py

```diff
@@ -28,9 +28,12 @@ class ConnectionStringParser:
 
     def __init__(
         self,
-        connection_string: str = None
+        connection_string: typing.Optional[str] = None,
+        *,
+        instrumentation_key: typing.Optional[str] = None,
```

Comment:
Why are you adding this?

Reply:
I added this based on the comments further down that implied this was a supported scenario (2nd in the priority heirarchy), but was missing. I can revert this and amend the comment if that is inaccurate?